tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -579,15 +579,21 @@ private static string GetApplicationName(TracerSettings settings)
         private static bool TryLoadAspNetSiteName(out string siteName)
         {
 #if NETFRAMEWORK
-            // System.Web.dll is only available on .NET Framework
-            if (System.Web.Hosting.HostingEnvironment.IsHosted)
+            try
+            {
+                // System.Web.dll is only available on .NET Framework
+                if (System.Web.Hosting.HostingEnvironment.IsHosted)
+                {
+                    // if this app is an ASP.NET application, return "SiteName/ApplicationVirtualPath".
+                    // note that ApplicationVirtualPath includes a leading slash.
+                    siteName = (System.Web.Hosting.HostingEnvironment.SiteName + System.Web.Hosting.HostingEnvironment.ApplicationVirtualPath).TrimEnd('/');
+                    return true;
+                }
+            }
+            catch (TypeLoadException)
```

Comment:
Is it worth including the exception in the log message? I'm not sure if it is, or if it would just confuse customers 🤷‍♂️